### DISCUSSION TITLE
[ideas] Propagate the condition of `tar_skip` in the downstream dependencies

### DISCUSSION THREAD

**Discussion by jospueyo** [ropensci/targets#1241] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'm interested in using `tar_skip()` to run conditionally two parts of the pipeline based on a setup value. I can use `tar_skip` in all targets of both branches, but it would nice if the condition was propagated downstream.

``` r
library(targets)
tar_script({
    library(tarchetypes)
    mode <- "b"
    list(
        tar_skip(mode_a, paste("mode", mode), skip = mode != "a"), 
        tar_skip(mode_b, paste("mode", mode), skip = mode != "b"), 
        tar_target(mode_a2, paste(mode_a, 2)), 
        tar_target(mode_b2, paste(mode_b, 2))
    )
})
tar_make()
#> ▶ dispatched target mode_b
#> ● completed target mode_b [0 seconds]
#> ▶ dispatched target mode_a
#> ■ canceled target mode_a
#> ▶ dispatched target mode_b2
#> ● completed target mode_b2 [0 seconds]
#> ▶ dispatched target mode_a2
#> ● completed target mode_a2 [0 seconds]
#> ▶ completed pipeline [0.16 seconds]
```

<sup>Created on 2024-02-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

The result of target `mode_a2` is `" 2"`. IMHO, this target should also be skipped because its upstream dependency was skipped.

**Comment by wlandau** [comment]
I have no plans to make skips propagate, but you can apply `tar_skip()` or `tar_cue_skip()`  to downstream targets.

**Comment by koefoeden** [comment]
Perhaps https://github.com/ropensci/targets/discussions/1423#discussioncomment-13320782 might be relevant for you
